xcdat/dataset.py

```diff
@@ -155,6 +161,84 @@ def open_mfdataset(
     return ds
 
 
+def infer_or_keep_var(
+    dataset: xr.Dataset, data_var: Optional[str] = None
+) -> xr.Dataset:
+    """Infer the data variable(s) or keep a specific one in the Dataset.
+
+    If ``data_var`` is None, then this function checks the number of
+    regular (non-bounds) data variables in the Dataset. If there is a single
+    regular data var, then it will add an 'xcdat_infer' attr pointing to it in
+    the Dataset. XCDAT APIs can call `get_inferred_var()` to get the data var
+    linked to the 'xcdat_infer' attr.
+
+    If ``data_var`` is not None, then this function checks if the ``data_var``
+    exists in the Dataset and if it is a regular data var. If those checks pass,
+    it will subset the Dataset to retain that ``data_var`` and any bounds data
+    vars. The 'xcdat_infer' attr  pointing to the data variable is also added
+    to the Dataset.
+
+    This utility function is useful for designing XCDAT APIs with an optional
+    ``data_var`` kwarg to allow implicit (``data_var=None``) or explicit
+    (``data_var='a_var'``) selection of a data var to perform an operation on.
```

Comment:
Change to:

    This utility function is useful for designing XCDAT APIs with an optional
    ``data_var`` kwarg. If ``data_var`` is None, an inference to the desired data var is performed
     with a call to this function. Otherwise, perform the API operation explicitly on ``data_var``.